common/lib/client/realtimechannel.js

```diff
@@ -28,6 +28,10 @@ var RealtimeChannel = (function() {
 			 * I.e. all encodings added by Ably Realtime have been decoded. */
 			baseEncodedPreviousPayload: undefined
 		};
+		this.lastPayloadMessageId = null;
+		this.lastPayloadProtocolMessageChannelSerial = null;
```

Comment:
You're storing three different bits of the last message in three different properties in the `channel` (`lastPayloadMessageId`, `lastPayloadProtocolMessageChannelSerial`, and `baseEncodedPreviousPayload` as part of the `encodingDecodingContext` (without which you wouldn't need keep that context as a property of the channel). Wouldn't it be a lot simpler & neater to just store the `lastMessage` wholesale?

Reply:
We discussed this shortly in a call with @paddybyers and reached the conclusion to leave it as is. `baseEncodedPreviousPayload` is decoded only to the point where `vcdiff` is removed or `base64` is removed, i.e. it is quite specific. Mutating  `lastMessage` to contain this kind of data is also not a good solution.